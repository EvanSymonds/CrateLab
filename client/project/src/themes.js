import { createMuiTheme, responsiveFontSizes } from "@material-ui/core/styles"
import red from "@material-ui/core/colors/red"
import grey from "@material-ui/core/colors/grey"


let redGreyTheme = createMuiTheme({
  palette: {
    primary: {
      main: red[600],
      contrastText: '#fff',
    },
    secondary: {
      main: grey[500],
      light: grey[300]
    },
  },
  typography:{
    button: {
      fontWeight: "500",
      textTransform: 'none',
    }
  }
})
redGreyTheme  = responsiveFontSizes(redGreyTheme)
export { redGreyTheme }