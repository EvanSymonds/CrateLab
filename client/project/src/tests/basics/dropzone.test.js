import React from 'react';
import { ThemeProvider } from "@material-ui/core/styles"
import { redGreyTheme } from "../../themes"
import Dropzone from "../../components/basics/dropzone"
import Enzyme, { mount } from "enzyme"
import EnzymeAdapter from "enzyme-adapter-react-16"
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new EnzymeAdapter()});

test("Renders a working dropzone", () => {

  const testDrop = jest.fn()

  const wrapper = mount(
    <ThemeProvider theme={redGreyTheme}>
      <Dropzone onDrop={testDrop()} disabled={false} />
    </ThemeProvider>
  )

  const d = wrapper.find("[data-test='component-dropzone']")
  
  d.prop("onDrop")

  expect(testDrop).toBeCalledTimes(1)

  const rendered = renderer.create(
    <ThemeProvider theme={redGreyTheme}>
      <Dropzone onDrop={testDrop()} disabled={false} />
    </ThemeProvider>
  )

  expect(rendered.toJSON()).toMatchSnapshot()
})