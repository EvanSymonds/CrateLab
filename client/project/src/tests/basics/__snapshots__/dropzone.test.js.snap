// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a working dropzone 1`] = `
<div
  className="makeStyles-root-39"
  data-test="component-dropzone"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 75,
      "borderStyle": "dashed",
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": 150,
      "justifyContent": "center",
      "margin": "30px",
      "width": 150,
    }
  }
  tabIndex={0}
>
  <div>
    <input
      disabled={false}
      id="icon-button-file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <label
      htmlFor="icon-button-file"
    >
      <div
        data-test="component-button"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root-55 MuiIconButton-root-46 makeStyles-root-42"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiIconButton-label-54"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-58 MuiSvgIcon-fontSizeLarge-66"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
              />
            </svg>
          </span>
        </span>
      </div>
    </label>
  </div>
  <div
    className="makeStyles-root-39"
  >
    Upload files
  </div>
</div>
`;
